### ISSUE TITLE
Firefox Container not using custom profile

### ISSUE THREAD

**Issue by M3ssman** [SeleniumHQ/docker-selenium#777]
## Meta -
Image(s):  
standalone-firefox, standalone-firefox-debug
Docker-Selenium Image Version:  
 3.14.0-beryllium
Docker Version:  
18.03.1-ce
OS: 
Ubuntu 16.04 LTS

## Expected Behavior -
Using a separate Firefox Profile (with JavaScript disabled, for example named  `dkj9e2nd.scriptless`) when starting container like explained at #359 and [How-to-use-a-Firefox-custom-profile](https://github.com/SeleniumHQ/docker-selenium/wiki/How-to-use-a-Firefox-custom-profile), i.e. map Profile Volume to  container and set docker env with JAVA_OPTS.

## Actual Behavior -
Inside Container Fallback to default temporary profile, as pointed out by container logs:

```
10:02:44.057 INFO [ActiveSessionFactory.apply] - Capabilities are: {
  "browserName": "firefox"
}
10:02:44.058 INFO [ActiveSessionFactory.lambda$apply$11] - Matched factory org.openqa.selenium.remote.server.ServicedSession$Factory (provider: org.openqa.selenium.firefox.GeckoDriverService)
1535536964060	geckodriver	INFO	geckodriver 0.21.0
1535536964066	geckodriver	INFO	Listening on 127.0.0.1:27968
1535536964074	mozrunner::runner	INFO	Running command: "/usr/bin/firefox" "-marionette" "-foreground" "-no-remote" "-profile" "/tmp/rust_mozprofile.g32MumTWqhGa"
1535536965480	Marionette	INFO	Listening on port 43391
10:02:45.631 INFO [ProtocolHandshake.createSession] - Detected dialect: W3C
10:02:45.648 INFO [RemoteSession$Factory.lambda$performHandshake$0] - Started new session b20f0359-91de-4729-aa78-05efe7abdaaf (org.openqa.selenium.firefox.GeckoDriverService)

```

## Steps to reproduce -

see Attachment (without Firefox profile, which is quite large)

**Comment by ddavison** [comment]
have you seen https://github.com/SeleniumHQ/docker-selenium/wiki/How-to-use-a-Firefox-custom-profile  ?

**Comment by M3ssman** [comment]
Ok, I tried, but I right now (62.0 (64-bit) Mozilla Firefox for Ubuntu canonical-1.0), I couldn't even start the ProfileManager as described with `firefox -P`. Opens a brand new window, instead.

**Comment by M3ssman** [comment]
Sorry, I didn' t get it right with ProfileManager in the first place since there was an instance of Firefox running somewhere. After killing all ff-instances, ProfileManager popped into place.

What I've done so far (guided by the wiki):

1. created fresh instance named `selenium`with setting `javascript.enabled`to value `false`. 
2. copied fresh profile from `/home/<user>/.mozilla/firefox/profiles`to project folder
3. run test assuming no javascript using this profile with my real, local browser (firefox 62.0). 
    tests run fine.
4. follow instructions to extend standalone-firefox container
    except: added JAVA_OPT to docker-compose environment section
    ```
    environment:
      - no_proxy=localhost
      - TZ=Europe/Berlin
      - JAVA_OPTS=-Dwebdriver.firefox.profile=selenium
    ```
    copied profile-data `aa3eqk4v.selenium` and profile.ini to `/home/seluser/.mozilla/firefox`
    `docker exec -it standalone-firefox-noscript ls -la /home/seluser/.mozilla/firefox` => 

   ```
   drwx------ 1 seluser seluser 4096 Sep 13 07:35 .
   drwx------ 1 seluser seluser 4096 Sep  5 14:43 ..  
   drwx------ 1 seluser seluser 4096 Sep 12 15:38 aa3eqk4v.selenium
   drwx------ 1 seluser seluser 4096 Sep  5 14:43 Crash Reports
   drwx------ 1 seluser seluser 4096 Sep  5 14:43 Pending Pings
   -rw-rw-r-- 1 seluser seluser  105 Sep 12 15:41 profiles.ini
   ```

When I use `selenium-side-runner`to execute tests against the container, it reveals 

 ```
standalone-firefox-noscript | 08:19:23.071 INFO [ActiveSessionFactory.apply] - Capabilities are: {
standalone-firefox-noscript |   "browserName": "firefox"
standalone-firefox-noscript | }
standalone-firefox-noscript | 08:19:23.072 INFO [ActiveSessionFactory.lambda$apply$11] - Matched factory org.openqa.selenium.remote.server.ServicedSession$Factory (provider: org.openqa.selenium.firefox.GeckoDriverService)
standalone-firefox-noscript | 1536819563091	geckodriver	INFO	geckodriver 0.21.0
standalone-firefox-noscript | 1536819563095	geckodriver	INFO	Listening on 127.0.0.1:31513
standalone-firefox-noscript | 1536819563293	mozrunner::runner	INFO	Running command: "/usr/bin/firefox" "-marionette" "-foreground" "-no-remote" "-profile" "/tmp/rust_mozprofile.DBU4G3cKAQwh"
standalone-firefox-noscript | 1536819564745	Marionette	INFO	Listening on port 34647
standalone-firefox-noscript | 08:19:24.905 INFO [ProtocolHandshake.createSession] - Detected dialect: W3C
standalone-firefox-noscript | 08:19:24.948 INFO [RemoteSession$Factory.lambda$performHandshake$0] - Started new session 31ae981e-6856-4d86-8435-6754d82c1264 (org.openqa.selenium.firefox.GeckoDriverService)

 ```

Looks like the container uses local tmp-Profile, instead of what has been provided.

When I run `echo $JAVA_OPTS`in interactive mode with the container, it prints out nothing. 
Does this mean the var in unknown inside container? 
??

**Comment by diemol** [comment]
Hi @M3ssman,

I also tried the instructions again, and even if the profile is correctly set, it seems that the Grid is not picking it when the property `-Dwebdriver.firefox.profile=selenium` is set. I think the issue happens because in the Grid the GeckoDriverService does not retrieve that property when GeckoDriver is started. Perhaps @barancev can confirm this?

I will leave this issue open while we find a way to fix it in the upstream project.

**Comment by barancev** [comment]
Yes, Selenium does not take into account `webdriver.firefox.profile` system property. Is it a regression issue?  Did it work in earlier versions?

**Comment by diemol** [comment]
@barancev well, this, and other properties, are present in the FirefoxDriver (which is legacy), so I am not sure if we expect to provide the same features now that the default driver is GeckoDriver.
